File: leetcode/217_contains_duplicate/main.py
def containsDuplicate(nums: list[int]) -> bool:
    dct = dict()

    for num in nums:
        if dct.get(num) == 1:
            return True
        else:
            dct[num] = 1

    return False



print(containsDuplicate([1, 2, 3, 4, 1]))
print(containsDuplicate([1, 2, 3, 4]))
print(containsDuplicate([1,1,1,1,3,3,4,3,2,4,2])) 

File: leetcode/217_contains_duplicate/main.rs
use std::collections::HashMap;

fn main() {
    println!("{}", contains_duplicate(&vec![1, 2, 3]));
    println!("{}", contains_duplicate(&vec![1, 2, 3, 1]));
    println!("{}", contains_duplicate(&vec![1, 1, 2, 3]));
}

fn contains_duplicate(nums: &Vec<u8>) -> bool {
    let mut hashmap: HashMap<u8, u8> = HashMap::new();
    
    for num in nums {
        match hashmap.get(num) {
            Some(_) => return true,
            None => {
                hashmap.insert(*num, 1)
            },
        };
    };

    false
}